File: s3GoSDK/sample/src/PutObject.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/s3"
)

func PutObject(arg [3]string, filePath [3]string) {
	file, err := os.Open(filePath[0])
	if err != nil {
		fmt.Println("Unable to open file " + filePath[0])
		return
	}
	defer file.Close()
	input := &s3.PutObjectInput{
		ACL:    "private",
		Bucket: aws.String(arg[0]),
		Key:    aws.String(filePath[0]),
		Body:   file,
	}
	_, putobject_err := Client.PutObject(context.TODO(), input)
	if putobject_err != nil {
		fmt.Println("Got error uploading file:")
		fmt.Println(putobject_err)
		return
	}
}

File: s3RustSDKExample/sample/src/put_object.rs
use s3::{ByteStream, Client};
use std::path::Path;
use std::process;

pub async fn putobject(client: Client, bucket: String, key: String) {
    let body = ByteStream::from_path(Path::new(&key))
        .await
        .unwrap_or_default();
    match client
        .put_object()
        .bucket(&bucket)
        .key(&key)
        .body(body)
        .send()
        .await
    {
        Ok(_) => {
            println!("Put object {} in {}\n", key, bucket);
        }
        Err(e) => {
            println!("Got an error put object");
            println!("{}", e);
            process::exit(1);
        }
    };
}
